purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -907,17 +891,23 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
     ) {
         val unsyncedSubscriberAttributesByKey =
             subscriberAttributesManager.getUnsyncedSubscriberAttributes(appUserID)
+        val productInfo = ProductInfo(
+            productID = purchase.sku,
+            offeringIdentifier = purchase.presentedOfferingIdentifier,
+            price = skuDetails?.priceAmount,
+            currency = skuDetails?.priceCurrencyCode,
+            duration = skuDetails?.subscriptionPeriod?.takeUnless { it.isEmpty() },
+            introDuration = skuDetails?.introductoryPricePeriod?.takeUnless { it.isEmpty() },
+            trialDuration = skuDetails?.freeTrialPeriod?.takeUnless { it.isEmpty() }
```

Comment:
it looks like we're getting a lot of data from skuDetails, maybe ProductInfo could have a constructor that takes skuDetails and read from there?

Reply:
I love the idea. Easier to test stuff too